### ISSUE TITLE
[aritmetic_side_effects] Arithmetic involving constant paramenters should not trigger the lint

### ISSUE THREAD

**Issue by c410-f3r** [rust-lang/rust-clippy#11266]
Rustc already handles this use-case.

```rust
#![deny(clippy::arithmetic_side_effects)]

pub fn foo<const N: usize>() {
    let _ = N + 1;
}

fn main() {
    foo::<{ usize::MAX }>();
}
```

**Comment by ijackson** [comment]
I think this could be solved by using symbolic execution to calculate possible ranges for values, and not producing the lint if the overflow can thereby be excluded.

Without something like that, expressions which have been carefully constructed to avoid overflow will produce this lint, requiring a manual suppression (and, presumably, tests and/or detailed review, to demonstrate the calculation is indeed overflow-free).
